### ISSUE TITLE
Unable to build the Miyoo release from source with translations

### ISSUE THREAD

**Issue by anton-vinogradov** [diasurgical/devilutionX#5628]
### Operating System

Linux x64/Miyoo

### DevilutionX version

1.4.1 (latest release)

### Describe

[Brocky's](https://github.com/Brocky/devilutionX) releases contain no translations, so I'm trying to compile Miyoo's release with translations by myself.
To enable translations I've added **poedit** installation to Miyoo's Dockerfile.

But, compilation fails on the following:
``` 
...
[ 97%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/paths.cpp.o
[ 97%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/pcx_to_clx.cpp.o
[ 97%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/sdl_bilinear_scale.cpp.o
[ 98%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/sdl_thread.cpp.o
[ 98%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/str_cat.cpp.o
[ 98%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/surface_to_clx.cpp.o
[ 99%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/utf8.cpp.o
[ 99%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/sdl2_to_1_2_backports.cpp.o
[ 99%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/engine/demomode.cpp.o
[ 99%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/effects.cpp.o
[100%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/engine/sound.cpp.o
[100%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/push_aulib_decoder.cpp.o
[100%] Building CXX object Source/CMakeFiles/libdevilutionx.dir/utils/soundsample.cpp.o
[100%] Built target libdevilutionx
Scanning dependencies of target devilutionx
[100%] Building CXX object CMakeFiles/devilutionx.dir/Source/main.cpp.o
[100%] Linking CXX executable devilutionx
/usr/bin/strip: Unable to recognise the format of the input file `/root/workspace/diablo/build-miyoo-mini/devilutionx'
make[2]: *** [CMakeFiles/devilutionx.dir/build.make:409: devilutionx] Error 1
make[2]: *** Deleting file 'devilutionx'
make[1]: *** [CMakeFiles/Makefile2:539: CMakeFiles/devilutionx.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
``` 

where build.make:409 is a
`/usr/bin/strip --strip-all /root/workspace/diablo/build-miyoo-mini/devilutionx`

Not sure if this is because of enabled translations, but any help is appreciated.

### To Reproduce

Follow the Miyoo's [installation guide](https://github.com/diasurgical/devilutionX/blob/master/docs/building.md) and (optionally?) install **poedit** before making the release.

### Expected Behavior

Successful compilation

**Comment by AJenbo** [comment]
`poedit` is a translation editor, what you need for building translations is `gettext`. The translations are separate files and are not system specific so you could just grab the .gmo files from another platform and add them to the Miyoo installation.

**Comment by AJenbo** [comment]
> where build.make:409 is a
> /usr/bin/strip --strip-all /root/workspace/diablo/build-miyoo-mini/devilutionx

This is just a step to minimize the size of the exec, the build was actually successful and should work. You can either try and find a strip program that is compatible with the file format used for Miyoo or just disable the strip process.

**Comment by glebm** [comment]
You're using host strip instead of the miyoo toolchain strip. Sounds like a bug with your (Miyoo's) CMake toolchain file.

**Comment by glebm** [comment]
Oh it's not even using a toolchain file, it's just overriding some env vars here: https://github.com/diasurgical/devilutionX/blob/master/Packaging/miyoo_mini/toolchain_env.sh

and setting some cmake variables here https://github.com/diasurgical/devilutionX/blob/efaf334965a641045a94d9c9c61592b0068c1a25/Packaging/miyoo_mini/build.sh#L23-L29

Short term, the latter should be updated with `-DCMAKE_STRIP=path/to/miyoo/strip`.

Really, this should be rewritten as a toolchain file to avoid all kinds of similar issues.

**Comment by anton-vinogradov** [comment]
Thanks a lot!
Will try to fix the compilation or grab translations.

> what you need for building translations is gettext
Could you please explain how to enable translations at compilation?

BTW, is it possible to release the Miyoo version here as well as all others?

**Comment by AJenbo** [comment]
> Could you please explain how to enable translations at compilation?

Install gettext. Without info about your system that is as specific as I can be.

**Comment by glebm** [comment]
None of us have a miyoo, this is a community-contributed port.
But if you show my comment to an experienced miyoo developer it will make perfect sense.
Perhaps direct someone from the miyoo community to this issue?

**Comment by glebm** [comment]
@anton-vinogradov I've sent a PR with fixes for the miyoo build.
Can you please test the resulting package? https://github.com/diasurgical/devilutionX/actions/runs/3775816206

**Comment by anton-vinogradov** [comment]
> @anton-vinogradov I've sent a PR with fixes for the miyoo build (#5631) Can you please test the resulting package? https://github.com/diasurgical/devilutionX/actions/runs/3775816206

Checked [devilutionx-miyoo-mini-onion-os.zip](https://github.com/diasurgical/devilutionX/suites/10048326216/artifacts/488929429) on the formatted card with fresh Onion 4.0.3, and It works!

Strange things:
1) Version is 1.4.0, not 1.4.1.
2) To change language from English to another you should select the language and restart the game, while [Brocky's](https://github.com/Brocky) release (with appended translations) changes the language immediately.

**Comment by glebm** [comment]
>  you should select the language and restart the game

Reproduced, it's a bug in the current master.

> Version is 1.4.0, not 1.4.1.

Yeah that is normal, 1.4.1 is built from a separate branch.
The actual version there is the current master (not yet released, will be bumped to 1.5.0 on release).
